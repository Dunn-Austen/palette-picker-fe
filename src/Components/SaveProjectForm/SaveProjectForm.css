.form-section {
  border: 1px solid black;
  height: 10vh;
  width: 400px;
}

.input-container {
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
}

label {
  font-size: 14px;
}

#project-input {
  height: 18px;
}

.project-form {
  align-items: center;
  display: flex;
  flex-direction: row;
  height: 10vh;
  justify-content: space-around;
}
