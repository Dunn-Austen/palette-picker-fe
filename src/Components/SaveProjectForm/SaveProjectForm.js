import React, {useState} from 'react';
import './SaveProjectForm.css';
import { postNewProject } from '../../apicalls';
import PropTypes from 'prop-types';

const SaveProjectForm = ({projects, setProjects}) => {

  let [inputValue, setInputValue] = useState('');

  const handleInputChange = (event) => {
    setInputValue(event.target.value);
  }

  const reformatResponseObject = (returnedBody) => {
    return {...returnedBody, palettes: [] }
  }

  return (
    <section className='form-section'>
      <form className='project-form'>
        <div className='input-container'>
          <label htmlFor='project-input'>Create a project: </label>
          <input
            id='project-input'
            type='text'
            placeholder='Kitchen Themes'
            onChange={handleInputChange}
            />
        </div>
        <button type='button' className='post-project'
          onClick={() => setProjects([...projects, reformatResponseObject(postNewProject({title: inputValue}))])}
        >
          Create Project
        </button>
      </form>
    </section>
  );
}

export default SaveProjectForm;

SaveProjectForm.propTypes = {
  projects: PropTypes.array,
  setProjects: PropTypes.func
}
