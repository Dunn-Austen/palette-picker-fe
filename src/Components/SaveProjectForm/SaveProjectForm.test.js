import React from 'react';
import { shallow } from 'enzyme';
import SaveProjectForm from './SaveProjectForm';

describe('SaveProjectForm', () => {
  it('should match the snapshot', () => {
    const mockProjects = [
      {
        id: 1,
        title: 'kitchen'
      }
    ];
    const mockAddProject = jest.fn();

    const wrapper = shallow(<SaveProjectForm 
      projects={mockProjects}
      addProject={mockAddProject}
    />);

    expect(wrapper).toMatchSnapshot();
  });
});
