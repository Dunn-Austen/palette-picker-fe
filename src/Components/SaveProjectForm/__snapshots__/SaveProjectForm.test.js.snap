// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SaveProjectForm should match the snapshot 1`] = `
<section
  className="form-section"
>
  <form
    className="project-form"
  >
    <div
      className="input-container"
    >
      <label
        htmlFor="project-input"
      >
        Create a project: 
      </label>
      <input
        className=""
        id="project-input"
        onChange={[Function]}
        placeholder="Kitchen Themes"
        type="text"
        value=""
      />
    </div>
    <button
      className="post-project"
      onClick={[Function]}
      type="button"
    >
      Create Project
    </button>
  </form>
</section>
`;
